broadinstitute/rawls
# [WOR-73] Initial create MC workspace API and service implementation

## Context
In #1629 we threaded through the new `workspace_type` from the DB to the associated models. Now, we need to wire up and conditionally create workspaces in WSM, and potentially related Azure cloud contexts.

## This PR
* Creates a new `MultiCloudWorkspaceService` which knows how to provision an azure workspace. For now, this codepath is limited to creating workspaces with Azure cloud contexts, attempts to call with `Gcp` will result in an exception
* Gates access to that method by requiring a new config setting `multiCloudWorkspaces.enabled` be set to `true`, _and_ that the incoming request has the `cloudPlatform` field set
* Adds tests for the new service
* **Note**: This is a PR against the aforementioned #1629. 
## TODO

____

  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
